fe/fe-core/src/main/java/com/starrocks/catalog/ColumnAccessPath.java

```diff
@@ -0,0 +1,111 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.catalog;
+
+import com.google.api.client.util.Lists;
+import com.starrocks.analysis.StringLiteral;
+import com.starrocks.thrift.TAccessPathType;
+import com.starrocks.thrift.TColumnAccessPath;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/*
+ * ColumnAccessPath is used to describe the access path of a complex(Map/Struct/Json) column.
+ *
+ * eg:
+ *  select struct_a.col_b.col_c.col_d
+ *
+ * ColumnAccessPath will be:
+ *  struct_a (ROOT)
+ *  -- col_b (FIELD)
+ *    -- col_c (FIELD)
+ *      -- col_d (FIELD)
+ *
+ * in complex sql, eg:
+ *  select  struct_a.col_g, struct_a.col_b.col_c.col_d, struct_a.col_b.col_e.col_f
+ * ColumnAccessPath will be:
+ *  struct_a (ROOT)
+ *  -- col_g (FIELD)
+ *  -- col_b (FIELD)
+ *    -- col_c (FIELD)
+ *      -- col_d (FIELD)
+ *    -- col_e (FIELD)
+ *      -- col_f (FIELD)
+ *
+ */
+public class ColumnAccessPath {
```

Comment:
`SubfieldAccessPath` better? Because your access paths are recorded by subfield level.

Reply:
yeah, you are right, but I think the path is bind to column, and use subfield on column is good, but use it on scan node is confusing ?